implementations/rust/ockam/ockam_node/src/storage/database/database_configuration.rs

```diff
@@ -108,12 +110,27 @@ impl DatabaseConfiguration {
     pub fn sqlite(path: &Path) -> DatabaseConfiguration {
         DatabaseConfiguration::Sqlite {
             path: Some(path.to_path_buf()),
+            single_connection: false,
         }
     }
 
     /// Create an in-memory sqlite configuration
     pub fn sqlite_in_memory() -> DatabaseConfiguration {
-        DatabaseConfiguration::Sqlite { path: None }
+        DatabaseConfiguration::Sqlite {
```

Comment:
>I would evolve the DatabaseConfiguration to introduce SqliteInMemory and SqliteOnDisk so that single_connection only makes sense when the path is Some.
👍 

> This way we won't have to AutoRetry when the database is used in memory only.
AFAIK, deadlocks can also occur with in-memory databases, so auto-retry should also be enabled in that case.